lib/improver/utilities/temporal.py

```diff
@@ -624,10 +625,21 @@ def process(self, cube_t0, cube_t1):
 
         time_list = self.construct_time_list(initial_time, final_time)
         cubes = iris.cube.CubeList([cube_t0, cube_t1])
-        cube = cubes.merge_cube()
+        cube = merge_cubes(cubes)
+
         interpolated_cube = cube.interpolate(time_list, iris.analysis.Linear())
+
+        # iris.analysis.Linear() modifies the dtype of time and forecast_period
+        # coords so need to revert back
+        dtype_time = cube_t0.coord('time').points.dtype
+        dtype_fp = cube_t0.coord('forecast_period').points.dtype
```

Comment:
Are we sure the forecast period will be the only aux_coord ever associated with time?  It might be better if this could be more flexible - ie find any coords on the input cube that are associated with the time dimension, and cast them all back to their original datatypes.

Reply:
Fair enough, as long as it is captured.